op-e2e/actions/kromampt_fork_test.go

```diff
@@ -0,0 +1,182 @@
+	gasPriceOracleMPTCodeHash    = common.HexToHash("0x3474b859129195c11a4b1aa14af951a2b3f83c7841aa5e73e67d55bd3379220c")
+)
+
+func TestKromaMPTNetworkUpgradeTransactions(gt *testing.T) {
+	t := NewDefaultTesting(gt)
+	dp := e2eutils.MakeDeployParams(t, defaultRollupTestParams)
+	genesisBlock := hexutil.Uint64(0)
+	kromaMPTOffset := hexutil.Uint64(4)
+
+	dp.DeployConfig.L1CancunTimeOffset = &genesisBlock // can be removed once Cancun on L1 is the default
+
+	// Activate all forks at genesis, and schedule KromaMPT the block after
+	dp.DeployConfig.L2GenesisRegolithTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisCanyonTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisDeltaTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisEcotoneTimeOffset = &genesisBlock
+	dp.DeployConfig.L2GenesisKromaMPTTimeOffset = &kromaMPTOffset
+	require.NoError(t, dp.DeployConfig.Check(), "must have valid config")
+
+	sd := e2eutils.Setup(t, dp, defaultAlloc)
+	log := testlog.Logger(t, log.LevelDebug)
+	_, _, _, sequencer, engine, verifier, _, _ := setupReorgTestActors(t, dp, sd, log)
+	ethCl := engine.EthClient()
+
+	// start op-nodes
+	sequencer.ActL2PipelineFull(t)
+	verifier.ActL2PipelineFull(t)
+
+	// Get gas price from oracle
+	gasPriceOracle, err := bindings.NewGasPriceOracleCaller(predeploys.GasPriceOracleAddr, ethCl)
+	require.NoError(t, err)
+
+	// Get fee vault recipients
+	protocolFeeVault, err := bindings.NewProtocolVaultCaller(predeploys.ProtocolVaultAddr, ethCl)
+	require.NoError(t, err)
+	protocolFeeRecipient, err := protocolFeeVault.RECIPIENT(nil)
+	require.NoError(t, err)
+
+	l1FeeVault, err := bindings.NewL1FeeVaultCaller(predeploys.L1FeeVaultAddr, ethCl)
+	require.NoError(t, err)
+	l1FeeRecipient, err := l1FeeVault.RECIPIENT(nil)
+	require.NoError(t, err)
+
+	// Get current implementations addresses (by slot) for L1Block + GasPriceOracle
+	initialL1BlockAddress, err := ethCl.StorageAt(context.Background(), predeploys.KromaL1BlockAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+	initialGasPriceOracleAddress, err := ethCl.StorageAt(context.Background(), predeploys.GasPriceOracleAddr, genesis.ImplementationSlot, nil)
+	require.NoError(t, err)
+
+	// Build to the pre-KromaMPT block
+	sequencer.ActBuildL2ToPreKromaMPT(t)
+
+	// get latest block
+	latestBlock, err := ethCl.BlockByNumber(context.Background(), nil)
+	require.NoError(t, err)
+	require.Equal(t, sequencer.L2Unsafe().Number, latestBlock.Number().Uint64())
+
+	transactions := latestBlock.Transactions()
+	// L1Block: 1 set-L1-info + 5 deploys + 5 upgradeTo + 1 enable KromaMPT on GPO
+	// See [derive.KromaMPTNetworkUpgradeTransactions]
+	require.Equal(t, 12, len(transactions))
+
+	// Use KromaDepositTx instead of DepositTx
+	for i, otx := range transactions {
+		if otx.Type() == types.DepositTxType {
+			transactions[i], err = otx.ToKromaDepositTx()
+			require.NoError(t, err)
+		}
+	}
```

Comment:
We don't need this conversion after Kroma MPT block, right?

Reply:
Yep that's right, but it seems that this conversion is not necessary for the success of the test. Do you mean by removing this?

Reply:
Yes I meant it's clear to remove it